src/engine/source/base/include/base/utils/threadEventDispatcher.hpp

```diff
@@ -30,22 +46,19 @@ template<typename T,
 class TThreadEventDispatcher
 {
 public:
-    explicit TThreadEventDispatcher(Functor functor,
-                                    const std::string& dbPath,
-                                    const uint64_t bulkSize = 1,
-                                    const size_t maxQueueSize = UNLIMITED_QUEUE_SIZE,
-                                    const uint8_t numberOfThreads = 1)
+    explicit TThreadEventDispatcher(
+        Functor functor,
+        const std::string& dbPath,
+        const uint64_t bulkSize = 1,
+        const size_t maxQueueSize = UNLIMITED_QUEUE_SIZE,
+        const ThreadEventDispatcherType dispatcherType = ThreadEventDispatcherType::SINGLE_THREADED_ORDERED)
         : m_functor {std::move(functor)}
         , m_maxQueueSize {maxQueueSize}
         , m_bulkSize {bulkSize}
         , m_queue {std::make_unique<TSafeQueueType>(TQueueType(dbPath))}
-        , m_numberOfThreads {numberOfThreads}
+        , m_numberOfThreads {dispatcherType == ThreadEventDispatcherType::MULTI_THREADED_UNORDERED ? MULTI_THREAD
```

Comment:
It is no longer necessary to maintain the number of threads in this variable if you already define it above.